config/main.py

```diff
@@ -1852,27 +1855,71 @@ def override_config_table(db, input_config_db, dry_run):
 
     config_db = db.cfgdb
 
+    # Read config from configDB
+    current_config = config_db.get_config()
+    # Serialize to the same format as json input
+    sonic_cfggen.FormatConverter.to_serialized(current_config)
+
+    updated_config = update_config(current_config, config_input)
+
+    yang_enabled = is_yang_config_verification_enabled(config_db)
+    if yang_enabled:
+        sy = sonic_yang_with_loaded_models(YANG_DIR)
+        # Validate running config
+        validate_config_by_yang(sy, current_config)
+        # Validate input config
+        validate_config_by_yang(sy, config_input)
+        # Validate updated whole config
+        validate_config_by_yang(sy, updated_config)
+
     if dry_run:
-        # Read config from configDB
-        current_config = config_db.get_config()
-        # Serialize to the same format as json input
-        sonic_cfggen.FormatConverter.to_serialized(current_config)
-        # Override current config with golden config
-        for table in config_input:
-            current_config[table] = config_input[table]
-        print(json.dumps(current_config, sort_keys=True,
+        print(json.dumps(updated_config, sort_keys=True,
                          indent=4, cls=minigraph_encoder))
     else:
-        # Deserialized golden config to DB recognized format
-        sonic_cfggen.FormatConverter.to_deserialized(config_input)
-        # Delete table from DB then mod_config to apply golden config
-        click.echo("Removing configDB overriden table first ...")
-        for table in config_input:
-            config_db.delete_table(table)
-        click.echo("Overriding input config to configDB ...")
-        data = sonic_cfggen.FormatConverter.output_to_db(config_input)
-        config_db.mod_config(data)
-        click.echo("Overriding completed. No service is restarted.")
+        override_config_db(config_db, config_input)
+
+
+def is_yang_config_verification_enabled(config_db):
+    device_metadata = config_db.get_entry('DEVICE_METADATA', 'localhost')
+    return 'enable' == device_metadata.get('yang_config_validation', None)
+
+
+def sonic_yang_with_loaded_models(yang_dir):
+    sy = sonic_yang.SonicYang(yang_dir)
+    sy.loadYangModel()
+    return sy
+
+
+def validate_config_by_yang(sy, config_json):
+    try:
+        tmp_config_json = copy.deepcopy(config_json)
+        sy.loadData(tmp_config_json)
+        sy.validate_data_tree()
+    except sonic_yang.SonicYangException as ex:
+        click.secho("Failed to validate config. Error: {}".format(ex), fg="red")
+        sys.exit(1)
```

Comment:
These 2 functions are for common purpose. Possible to reuse SonicYangExtMixin or extend libraries?

Reply:
You mean `loadData` and `validate_data_tree`? If so, I think it is not necessary to combine two into one.

Reply:
I do not mean to combine two functions into one. Is it possible to move these functions to a common library and make it reusable? I see they are similar to the features in `SonicYangExtMixin`.

Reply:
Test by using class ConfigMgmt, there are some unwanted logs:
```
def validate_config_by_yang(sy, config_json, jname):
    tmp_config_json = copy.deepcopy(config_json)
    cm = ConfigMgmt()
    try:
        cm.loadData(tmp_config_json)
        cm.validateConfigData():
    except Exception as ex:
         click.secho("Failed to validate {}. Error: {}".format(jname, ex), fg="magenta")
         sys.exit(1)
---below 1) and 2) together are whole print---
  ---1) Default class print---
    sonic_yang(6):Note: Below table(s) have no YANG models: CONSOLE_SWITCH, DEVICE_NEIGHBOR_METADATA, DHCP_SERVER, RESTAPI, SNMP, SNMP_COMMUNITY
    sonic_yang(6):Note: Below table(s) have no YANG models: CONSOLE_SWITCH, DEVICE_NEIGHBOR_METADATA, DHCP_SERVER, RESTAPI, SNMP, SNMP_COMMUNITY
    sonic_yang(6):Note: Below table(s) have no YANG models: CONSOLE_SWITCH, DEVICE_NEIGHBOR_METADATA, DHCP_SERVER, RESTAPI, SNMP, SNMP_COMMUNITY
    libyang[0]: Invalid value "denable" in "synchronous_mode" element. (path: /sonic-device_metadata:sonic-device_metadata/DEVICE_METADATA/localhost/synchronous_mode)
    sonic_yang(3):Data Loading Failed:Invalid value "denable" in "synchronous_mode" element.
  ---2) My Print---
    Failed to validate config_input. Error: Data Loading Failed
    Invalid value "denable" in "synchronous_mode" element.
```

Reply:
Let's track the verbose output issue as a github issue at low priority.

Reply:
Done. https://github.com/sonic-net/sonic-utilities/issues/2322